### DISCUSSION TITLE
Increased watch battery by xDrip complication?

### DISCUSSION THREAD

**Discussion by cwpeterson** [NightscoutFoundation/xDrip#2026] [Q&A]
Over the past couple months I've noticed a sporadic spike in battery usage by the xDrip complication. It's not consistent day to day so not sure what is causing this. I am on 2122021518 nightly right now and the watch is updated to match. Today my watch was at 40% after 4 hours of use. I had to switch to extended mode to make it through the day.

At 4:47pm you can see the attached screen shot I took of battery usage. Sometimes if I reboot the watch and charge it seems better the next day. 

My Fossil Gen 5 is a couple years old and definitely not getting the battery life it once did. I only have bluetooth, tilt to wake, and location enabled on it right now.

Anyone have any ideas why the xDrip complication would eat up this much battery?

![Screenshot_20220301-164711](https://user-images.githubusercontent.com/12939749/156285949-5dd1c058-fe0c-4101-897a-3fc14df63cd3.png)

**Comment by imakeinterfaces** [comment]
I would periodically run into this on another fossil product, the misfit vapor X. I would first unpairing and reinstalling the OS on the watch as this usually fixed it for me. My ticwatch also had a similar issue over time. I eventually got frustrated with wear OS and switched to Garmin. Now I get 4+ days of battery on my vivoactive 3. Given my experience I really don't think its the bluetooth that's eating the battery, its something else. The burndown shows exactly that.

**Reply by cwpeterson** [reply]
Yeah it's frustratingly inconsistent. Today I'm having one of the best battery days in months.  Nearly 8 hours on battery and still 65% remaining and xDrip used almost nothing. So strange


![Screenshot_20220304-154128](https://user-images.githubusercontent.com/12939749/156845733-a3acdf6a-8a88-475b-ac52-2c491c72d217.png)
.

**Reply by imakeinterfaces** [reply]
Oh also i'm assuming you have the watch collector modes turned off in xdrip prefs on the watch right? Those don't really work on fossil wearOS (on sony sw3 and a few others) and will run down the battery if enabled.  IIRC there's one that's basically "use watch as collector if phone cannot be reached" Make sure that is turned OFF.

**Reply by cwpeterson** [reply]
Yes everything is disabled since I don't use the watch as a collector. Just a display.

**Reply by utzaki** [reply]
I also can confirm, that the collector mode is disabled. I am quite sure the problem appeared after one of the recent updates.

**Reply by ojalaj** [reply]
Sorry to bump into this thread after a year+, but I'm having this exact issue with my Samsung GW4. I can't find the watch collector mode setting in XDrip app in watch - where is it? Is there anything else new on this issue?

**Comment by Navid200** [comment]
How do you know exactly how much longer the battery can run?
Is everything based on a voltage monitor?
Or, have you allowed the battery run until it stops working?

**Reply by cwpeterson** [reply]
That info comes directly from the WearOS app.

**Reply by cwpeterson** [reply]
I don't know how accurate it is, but today it's killing it for battery time.

**Comment by utzaki** [comment]
I have the same experience. After factory reset and reinstalling everything there is no battery drain for some days, but soon the problem is back. Some days are good, others bad.
I gave up some days ago and stopped using xDrip on my wear. 
Huawei Watch 2, Watchmaker with xDrip complications. Xdrip 2021.11.06

**Reply by cwpeterson** [reply]
It's odd, mine has been really good this week now. Even made one full long day with around 20% to spare without having to charge.

**Comment by oNSXo** [comment]
I'm seeing similar results with my Galaxy Watch 4.  Can't recall when it exactly started, but I believe some time this year.

![Screenshot_20220322-060033_Galaxy Watch4 Manager](https://user-images.githubusercontent.com/34994438/159480817-bdb917bd-b784-48f2-9b0f-f8764dcceb30.jpg)
![Screenshot_20220322-060027_Galaxy Watch4 Manager](https://user-images.githubusercontent.com/34994438/159480820-416432da-ac6b-4682-b764-32848b3a837c.jpg)
.

**Reply by cwpeterson** [reply]
Mine has actually been great for the last couple weeks. Better than it's been in months, but today I just noticed it's almost dead and I've only had it off charger about 4 hours.

**Reply by ojalaj** [reply]
@oNSXo did you ever sort this out?

**Reply by oNSXo** [reply]
@ojalaj  I'm still going strong on what I posted below since there seemed to be no movement here.  It uses Tasker to scrape xDrip's notification and pushes it to a new complication on the watch.  Have had zero issues w/ it for over a year now.

https://github.com/NightscoutFoundation/xDrip/discussions/2026#discussioncomment-3910091

**Reply by ojalaj** [reply]
Great - thanks! Is it the WearTasker from Google Store? Could you give some brief step-by-step instructions how to get it working.

**Reply by oNSXo** [reply]
It's "Tasker" in the Play Store (https://play.google.com/store/apps/details?id=net.dinglisch.android.taskerm&hl=en_US&gl=US)

There are 2 required plugins:

- [AutoNotification](https://play.google.com/store/apps/details?id=com.joaomgcd.autonotification&hl=en_US&gl=US)
- [AutoWear](https://play.google.com/store/apps/details?id=com.joaomgcd.autowear&hl=en_US&gl=US)

I had a different version that had a bit less information on it that I'll link to where I typed up some instructions about how to use it.  Same steps would apply to either profile:
https://gist.github.com/oNSXo/0706201d3ac28a99e1855d772e3039eb?permalink_comment_id=4351085#gistcomment-4351085

**Reply by oNSXo** [reply]
One step additionally that I didn't include is that the XML file gist needs to be saved as an XML file (ex. xDrip-complication.Tasker.prf.xml) on your device.  Then when you open Tasker, long press on the "Profiles" tab, Import Profile, and select that XML file.

**Reply by ojalaj** [reply]
@oNSXo Thanks for the reply! I'll hold on for another moment until adding another layer of apps for this. I opened another discussion thread for this. Hopefully developers and/or other experts would jump in.

**Reply by egyptianbman** [reply]
Thanks for sharing that. I've been fighting my watch battery for weeks and finally decided to look it up and found this thread. This was the first night in a looong time that I didn't wake up with a dead watch battery :D

**Comment by asavageiv** [comment]
I also have a Fossil Gen 5 and suffer from poor battery life with xDrip+. It does also seem sporadic. I don't really need xDrip to push new values to the watch every 5 minutes. I wonder if the problem could be solved with a mode where it pulls the data from the phone on command or on watch face wakeup or something like that.

**Reply by oNSXo** [reply]
I've pinned my app to 2022.01.31 and turned off updates since there's been no traffic on this issue.  No more issues w/ battery drain.

**Reply by cwpeterson** [reply]
I'm running 2022.07.26 nightly and it's been pretty good since. For a while I had to have everything off including display to make it through the day. Fossil's latest update also caused some issues by taking away my custom battery settings, but I found how to change things to how I prefer them. Sadly my watch battery is only good for maybe 12 hours unless I disable wake on tilt. But XDrip has been better for usage since I updated.

**Reply by asavageiv** [reply]
Thanks for this post. It does seem to have improved since updating to 22022.08.03. We'll see if it stays that way

**Reply by asavageiv** [reply]
Nope. The same issue returned. I'll try 2022.01.31

**Reply by oNSXo** [reply]
Make sure you downgrade your watch app version as well, otherwise it'll constantly try to update itself since it checks if the versions match.

**Comment by kssxs** [comment]
Hi all,

I've noticed that xDrip Prefs uses more battery on the watch as well, more so than anything else (while being used as a complication).  From my attached screenshot, you can see it's used about 6% so far, which isn't much, but compared to everything else, doesn't make sense.  I'm not using the BT Collector on the watch, it's only functioning as a BG display from the phone.  It really shouldn't be using even that amount of battery, right?  Are there additional settings changes that could reduce that amount, or is just a build issue?  I've got the last official release of May 19, 2022 on both phone and watch.  I'm using a new Samsung Watch 5 Pro, and this screenshot includes a major watch update that came down as well, so overall battery usage isn't quite accurate.

Should I downgrade to 2022.01.31 on both devices as well, or am I being overly zealous?

![Samsung Galaxy Watch 5 Pro - xDrip Battery Usage](https://user-images.githubusercontent.com/112791442/188788398-6efc19d2-452d-4948-a751-e37da183456f.jpg)

**Reply by cwpeterson** [reply]
I've found when I see it's starting to use a lot of battery rebooting the watch will help. For a few days at least.

**Comment by kssxs** [comment]
I'm hoping the Dev or a contributor can jump in and advise... a majority of battery usage shouldn't be taken up by updating a few numbers on a complication (I realize it's not as simple as that, but you get my drift).  My battery usage is relatively minor, but I'm thinking it shouldn't be more than 0.5 - 1% based on the usage by other components of the OS and apps.

@jamorham or @Navid200 - any ideas here guys?  Sorry if I'm breaching protocol to ask you guys specifically...  Can this issue be looked at please?  It's obviously a concern for those of us who are diabetics and using smartwatches and xDrip+ .  Thank you!

**Reply by cwpeterson** [reply]
I don't disagree. I just checked mine and my watch is at 63% with XDrip Prefs using 1% today. I put it on at approximately 7:40am and it's 2:45 pm. Says I have 12 hours left, which I know is wrong since the watch typically dies when the battery gets below 15%. I'm ok with this for now. I'm looking forward to the Pixel Watch announcement and hoping that will be my next smart watch. Just as long as I can run XDrip complication on it!

**Reply by Navid200** [reply]
> @jamorham or @Navid200 - any ideas here guys? 

I don't know much about watches and specially watch complications.  Isn't that (watch complication) something written by someone else outside this repository?

**Reply by oNSXo** [reply]
> > @jamorham or @Navid200 - any ideas here guys?
> 
> I don't know much about watches and specially watch complications. Isn't that (watch complication) something written by someone else outside this repository?

@Navid200 This is the only CGM related APK that I've got on either my phone or watch, so I'm pretty certain this complication is from the xDrip repo.  I continue to stay on 2022.01.31 as any newer version eats battery like crazy.

**Reply by ojalaj** [reply]
@cwpeterson How did it ultimately go with you on this issue?

**Comment by oNSXo** [comment]
FWIW if you've got (or are willing to get) Tasker along w/ the AutoNotification and AutoWear plugins, I came up w/ a simple profile that'll scrape xDrip's persistent notification and update an AutoWear complication with the data.  Allows me to update to the latest version of xDrip and remove the xDrip wear app due to it's significant amount of battery utilization on the watch.

https://gist.github.com/oNSXo/470b72ad8f6595a71bf4b52c408d217a

**Reply by cwpeterson** [reply]
Interesting. I won a Pixel watch, waiting for it to ship. I wonder if I'll be able to run the xDrip complication on it? We'll see. Last 2 days my battery on my Fossil Gen 5 started draining like crazy again and I can't determine why. The battery status don't show that much usage from any specific app, but I've had to disable tilt to wake just to make it through the work day. Once I get the Pixel Watch I'll try it out and test out your Tasker function.

**Reply by oNSXo** [reply]
Yeah, I was still pinned to the Jan version and I could barely get 24 hours out of my Galaxy Watch 4 and the battery status pointed to "xDrip Prefs".  No idea why it started doing that.  Now, after 24 hours I still have >50% battery life left.

That's also one reason I'm fine w/ the complication just giving the actual date instead of trying to be accurate about how many minutes in the past it updated (I'm sure that drains more battery updating at least once a minute).

Looks like the Pixel runs Wear OS, so I'd think it should work no problem.  Curious to see what you think about it; obviously, pretty limited on the information we can grab since it's limited to what's on the notification.

**Reply by belfastraven** [reply]
> Interesting. I won a Pixel watch, waiting for it to ship. I wonder if I'll be able to run the xDrip complication on it? We'll see. Last 2 days my battery on my Fossil Gen 5 started draining like crazy again and I can't determine why. The battery status don't show that much usage from any specific app, but I've had to disable tilt to wake just to make it through the work day. Once I get the Pixel Watch I'll try it out and test out your Tasker function.

I am running  with the xdrip complication on my new google watch.  In order to install xdrip on the watch,  I had to use the wear installer2 app.  When xdrip itself tried, it seemed to have a problem recognizing  the wear version on the watch.    I am running the latest nightly of xdrip+ on a pixel7pro  and actually using my bluejay watch as the collector.  In a couiple of days I will try to use the pixel watch as the collector.   I have to use a dexcom receiver for insurance purposes,  so whatever I use as a collector needs to run on the watch slot... .  Not sure yet how to see which apps are using battery on the watch....

**Reply by cwpeterson** [reply]
Got my new watch and am still playing with it and feeling it out. Haven't tried to install XDrip prefs/complication yet. Question for @oNSXo on your tasker integration. Are you able to change the look so it's a standard round complication? I personally just like to see current blood sugar and then the delta so I have an idea of the trend. I don't have the arrow on mine and I never look at last updated on the watch unless it's timed out due to disconnection or during warm up of a new sensor.

**Reply by oNSXo** [reply]
@cwpeterson Sure, looks fine still w/ the trend arrow, could probably take that out if you really don't want it though.

https://gist.github.com/oNSXo/0706201d3ac28a99e1855d772e3039eb

![Screenshot_20221026_174650_sysui](https://user-images.githubusercontent.com/34994438/198153503-adc63998-aecc-4cb7-9b0c-d044abe7a25b.png)

**Reply by cwpeterson** [reply]
@oNSXo Nice! I will probably play with it some. I'm a tad confused on the arrangement of the numbers. On my Fossil watch using the XDrip complication, blood sugar is on top, arrow to the right with the delta below. I did get it side loaded to test on my Pixel watch and it's inverted. Delta on top, blood sugar and trend arrow on the bottom. I can't figure out if there is a way to re-arrange them. I see on your watch face delta is also above blood sugar.
